src/main/java/de/tum/in/www1/artemis/service/connectors/pyris/PyrisHealthIndicator.java

```diff
@@ -0,0 +1,47 @@
+package de.tum.in.www1.artemis.service.connectors.pyris;
+
+import java.net.URI;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.actuate.health.Health;
+import org.springframework.boot.actuate.health.HealthIndicator;
+import org.springframework.context.annotation.Profile;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.service.connectors.ConnectorHealth;
+import de.tum.in.www1.artemis.service.connectors.pyris.dto.PyrisHealthStatusDTO;
+
+@Component
+@Profile("iris")
+public class PyrisHealthIndicator implements HealthIndicator {
+
+    private final RestTemplate restTemplate;
+
+    @Value("${artemis.iris.url}")
+    private URI irisUrl;
+
+    public PyrisHealthIndicator(@Qualifier("shortTimeoutPyrisRestTemplate") RestTemplate restTemplate) {
+        this.restTemplate = restTemplate;
+    }
+
+    /**
+     * Ping Iris at /health and check if the service is available and what its status is.
+     */
+    @Override
+    public Health health() {
+        ConnectorHealth health;
```

Comment:
I would only like to activate this on production, if this is really stable. If I would receive again many emails with "Artemis is down", I rather would like to avoid this health end point

Reply:
Yes this new health indicator is now stable, as it now only checks if Pyris is actually available, not if the models of Pyris are healthy, which previously caused the flakyness.